plugin-server/src/main/ingestion-queues/buffer.ts

```diff
@@ -0,0 +1,75 @@
+import Piscina from '@posthog/piscina'
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { Hub } from '../../types'
+import { runInstrumentedFunction } from '../utils'
+
+export function runBufferEventPipeline(hub: Hub, piscina: Piscina, event: PluginEvent) {
+    hub.lastActivity = new Date().valueOf()
+    hub.lastActivityType = 'runBufferEventPipeline'
+    return piscina.run({ task: 'runBufferEventPipeline', args: { event } })
+}
+
+export async function runBuffer(hub: Hub, piscina: Piscina): Promise<void> {
+    let eventRows: { id: number; event: PluginEvent }[] = []
+    await hub.db.postgresTransaction(async (client) => {
+        const eventsResult = await client.query(`
```

Comment:
This should be in db.ts.

Also we have now zero standard instrumentation on this query - impossible to tell when it's causing us issues. We enforce db.postgresQuery instead of `client.query` for that reason everywhere.